redis/src/test/java/org/hibernate/ogm/datastore/redis/impl/TestRedisDatastoreProvider.java

```diff
@@ -37,8 +39,26 @@ public void run() {
 			staticInstance = redisClient;
 		}
 
-		return staticInstance;
+		return (RedisClient) staticInstance;
+	}
+
+	@Override
+	protected RedisClusterClient createClusterClient(Hosts hosts) {
+		if ( staticInstance == null ) {
+			final RedisClusterClient redisClient = super.createClusterClient( hosts );
+
+			Runtime.getRuntime().addShutdownHook(
+					new Thread( "RedisClient shutdown hook" ) {
```

Comment:
The `DatastoreProvider#stop()` is called after each test and we want to keep the `RedisClient` instances to prevent long test durations caused by the lengthy client shutdown.